torch/utils/backend_registration.py

```diff
@@ -47,6 +47,15 @@ def rename_privateuse1_backend(backend_name: str) -> None:
     (2) manual_seed_all(seed: int) -> None
         Sets the seed for generating random numbers for your devices.
 
+    (3) device_count() -> int:
```

Comment:
Not for this PR, but if we want a nicer way to share these details, we can definitely add a new note (like https://pytorch.org/docs/stable/notes/extending.html) or tutorial just about this!

Reply:
yeah, @albanD , I am moving all these comment and note to the `extending.html`, and I will give a PR recently.

Reply:
btw I want to break down extending.html into smaller more manageable chunks.
Can you put all of these in a separate note that we reference from extending.html?

Reply:
@albanD yeah,  it is ok. Could you give a recommended location to put these notes？